sormas-backend/src/main/java/de/symeda/sormas/backend/util/PatchHelper.java

```diff
@@ -0,0 +1,335 @@
+	 * If the existingObject contains inner objects:
+	 * - a) of type HasUuid - those can be either updated or replaced.
+	 * In order to be updated the uuid of the inner object has to be provided and has to be the same as the existing one.
+	 * If the uuid differs the inner object is replaced.
+	 * - b) not of type HasUuid - are replaced.
+	 * !The logic if any inner object or value is allowed to be replaced or not relies entirely on the application!
+	 * 
+	 * THIS METHOD IS EXPERIMENTAL!!!!
+	 * 
+	 * @param jsonObject
+	 *            - the json payload containing the fields that are to be replaced. The structure of the json matches the structure of the
+	 *            Dto given by the existingObject
+	 * @param existingObject
+	 *            - the existing object whose fields are to be updated
+	 * @param <T>
+	 *            - the type of the object
+	 */
+	@Experimental
+	public static <T extends HasUuid> void postUpdate(@NotNull JsonNode jsonObject, @NotNull T existingObject) {
+		String objectUUID = existingObject.getUuid();
+		String jsonObjectUUID = jsonObject.get(EntityDto.UUID).textValue();
+		if (!objectUUID.equals(jsonObjectUUID)) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.patchWrongUuid, jsonObjectUUID, objectUUID));
+		}
+		postUpdateOrReplace(jsonObject, existingObject);
+	}
+
+	private static <T extends HasUuid> void postUpdateOrReplace(@NotNull JsonNode jsonObject, @NotNull T existingObject) {
+
+		Iterator<Map.Entry<String, JsonNode>> jsonObjectFieldMap = jsonObject.fields();
+		while (jsonObjectFieldMap.hasNext()) {
+			Map.Entry<String, JsonNode> jsonObjectFieldMapEntry = jsonObjectFieldMap.next();
+			JsonNode jsonObjectFieldNode = jsonObjectFieldMapEntry.getValue();
+			Field existingObjectField = getClassFieldByName(jsonObjectFieldMapEntry.getKey(), existingObject.getClass());
+			existingObjectField.setAccessible(true);
+
+			if (jsonObjectFieldNode.isObject()) {
+				updateExistingObject(existingObject, jsonObjectFieldNode, existingObjectField);
+
+			} else if (jsonObjectFieldNode.isArray()) {
+				updateObjectList(existingObject, jsonObjectFieldNode, existingObjectField);
+
+			} else {
+				setFieldValue(existingObject, existingObjectField, jsonObjectFieldNode);
+			}
+		}
+	}
+
+	/**
+	 * This method updates the fields of type collections (lists or array).
+	 * 
+	 * @param existingObject
+	 * @param jsonObjectFieldNode
+	 * @param existingObjectField
+	 * @param <T>
+	 * @throws JsonProcessingException
+	 * @throws IllegalAccessException
+	 */
+	private static <T extends HasUuid> void updateObjectList(T existingObject, JsonNode jsonObjectFieldNode, Field existingObjectField) {
+		Class listElementClass = getParameterizedType(existingObjectField);
+		List tempNewElementList = new ArrayList();
+
+		for (JsonNode listElement : jsonObjectFieldNode) {
+			T updatedObjectList = createOrUpdateElement(existingObject, existingObjectField, listElementClass, listElement);
```

Comment:
Just for my understanding, the element returned by createOrUpdateElement here is not a list, but an object that is part of the list, right? In that case, the variable should rather be called updatedListObject for clarification; if I'm mistaken, ignore this suggestion